src/nbl/video/COpenGLFence.h

```diff
@@ -48,14 +48,29 @@ class COpenGLFence final : public IGPUFence, public IOpenGLSyncPrimitiveBase
             return ES_SUCCESS;
     }
 
+    E_STATUS getStatus(IOpenGL_FunctionTable* _gl)
+    {
+        if (!isWaitable())
+            return ES_NOT_READY;
+        auto status = m_sync->waitCPU(_gl, 0ull);
+        switch (status)
+        {
+        case COpenGLSync::ES_TIMEOUT_EXPIRED:
+            return ES_NOT_READY;
+        case COpenGLSync::ES_FAIL:
+            return ES_ERROR;
+        default:
+            return ES_SUCCESS;
+        }
+    }
```

Comment:
are you planning on anything that derives from `IOpenGLSyncPrimitiveBase` such as `COpenGLFence` to be only touchable in a Queue/Swapchain thread (thread with a valid GL context)?

Reply:
? hm? like how? I mean they are only touchable by threads with GL ctx. No other threads has GL function table. Not sure I get what I mean

Reply:
ah ok

> No other threads has GL function table

You should probably laser that into the documentation/wiki/my-forehead